src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,413 @@
+        /// </summary>
+        /// <param name="serialPortStream">A stream to an Arduino/Firmata device</param>
+        public ArduinoBoard(Stream serialPortStream)
+        {
+            _dataStream = serialPortStream;
+            _spiEnabled = 0;
+            _supportedPinConfigurations = new List<SupportedPinConfiguration>();
+            FirmwareName = string.Empty;
+            FirmataVersion = new Version();
+            FirmwareVersion = new Version();
+        }
+
+        /// <summary>
+        /// Creates an instance of the Arduino board connection connected to a serial port
+        /// </summary>
+        /// <param name="portName">Port name. On Windows, this is usually "COM3" or "COM4" for an Arduino attached via USB.
+        /// On Linux, possible values include "/dev/ttyAMA0", "/dev/serial0", "/dev/ttyUSB1", etc.</param>
+        /// <param name="baudRate">Baudrate to use. It is recommended to use at least 115200 Baud.</param>
+        public ArduinoBoard(string portName, int baudRate)
+        {
+            _serialPort = new SerialPort(portName, baudRate);
+            _serialPort.Open();
+            _dataStream = _serialPort.BaseStream;
+            _supportedPinConfigurations = new List<SupportedPinConfiguration>();
+            FirmwareName = string.Empty;
+            FirmataVersion = new Version();
+            FirmwareVersion = new Version();
+        }
+
+        /// <summary>
+        /// Default pin numbering scheme of this board. Always <see cref="PinNumberingScheme.Logical"/>, because the Arduino has only one pin numbering scheme.
+        /// </summary>
+        public PinNumberingScheme DefaultPinNumberingScheme
+        {
+            get
+            {
+                // We have only one scheme
+                return PinNumberingScheme.Logical;
+            }
+        }
+
+        /// <summary>
+        /// Attach to this event to retrieve log messages. The Exception argument may be null if it is only an informational message.
+        /// </summary>
+        public event Action<string, Exception?>? LogMessages;
+
+        /// <summary>
+        /// Searches the given list of com ports for a firmata device.
+        /// </summary>
+        /// <remarks>
+        /// Scanning ports and testing for devices may affect unrelated devices. It is advisable to exclude ports known to contain other hardware from this scan.
+        /// A board won't be found if its port is already open (by the same or a different process).
+        /// </remarks>
+        /// <param name="comPorts">List of com ports. Can be used with <see cref="SerialPort.GetPortNames"/>.</param>
+        /// <param name="baudRates">List of baud rates to test. <see cref="CommonBaudRates"/>.</param>
+        /// <returns>A board, already open and initialized. Null if none was found.</returns>
+        public static ArduinoBoard? FindBoard(IEnumerable<string> comPorts, IEnumerable<int> baudRates)
```

Comment:
Whats the use case of having this overload public? Won't people always want to call the parameterless one? when would somebody would want to manually pass in ports and baudRates?

Reply:
There are multiple possible reasons to use this: You might want to scan only a set of ports, because you know which ports are certainly wrong and should not be probed (see also comment). And you might be using a non-default baudrate, or more likely, know the correct baudrate. I use this overload in tests to find the board, but with a known baudrate.